go/worker/storage/worker.go

```diff
@@ -39,9 +39,12 @@ func New(
 	registration *registration.Worker,
 ) (*Worker, error) {
 	enabled := config.GlobalConfig.Mode.HasLocalStorage()
-	if config.GlobalConfig.Mode == config.ModeArchive && len(commonWorker.GetRuntimes()) > 0 {
+	if config.GlobalConfig.Mode == config.ModeArchive {
```

Comment:
Should `HasLocalStorage` return true for archive mode? According to `registry.go`, the answer should be yes.

`haveLocalStorageWorker := config.GlobalConfig.Mode.HasLocalStorage() && config.GlobalConfig.Mode != config.ModeArchive`

Reply:
Good point, let me refactor a bit.